src/ad.jl

```diff
@@ -31,7 +42,7 @@ for algType in (Bisection, Brent, Alefeld, Falsi, ITP, Ridder)
 end
 
 function __nlsolve_ad(
-        prob::Union{IntervalNonlinearProblem, NonlinearProblem}, alg, args...; kwargs...)
+        prob::Union{IntervalNonlinearProblem, ImmutableNonlinearProblem}, alg, args...; kwargs...)
```

Comment:
```suggestion
        prob::Union{IntervalNonlinearProblem, NonlinearProblem, ImmutableNonlinearProblem}, alg, args...; kwargs...)
```

Reply:
This needs to happen in order for downstream since it hits this dispatch directly.